src/lematerial_forgebench/metrics/conditional_generation.py

```diff
@@ -0,0 +1,586 @@
+from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+
+from lematerial_forgebench.metrics.base import BaseMetric, MetricConfig
+from lematerial_forgebench.metrics.hhi_metrics import (
+    HHIProductionMetric,
+    HHIReserveMetric,
+)
+from lematerial_forgebench.utils.logging import logger
+
+
+@dataclass(kw_only=True)
+class DiscreteTargetConfig(MetricConfig):
+    """Configuration for the DiscreteTarget metric.
+
+    Parameters
+    ----------
+    target_value : Any
+        The target discrete value to match.
+    """
+
+    target_value: Any
+
+
+class DiscreteTargetMetric(BaseMetric):
+    """Generic metric for evaluating if structures match a target discrete value.
+
+    Parameters
+    ----------
+    target_value : Any
+        The target discrete value to match.
+    lower_is_better : bool, default=True
+        Whether a lower value is better.
+    name : str, default=None
+        The name of the metric.
+    description : str, default=None
+        The description of the metric.
+    n_jobs : int, default=1
+        The number of jobs to run in parallel.
+    """
+
+    def __init__(
+        self,
+        target_value: Any,
+        lower_is_better: bool = True,
+        name: str | None = None,
+        description: str | None = None,
+        n_jobs: int = 1,
+    ):
+        super().__init__(
+            name=name or f"DiscreteTarget_{str(target_value)}",
+            description=description
+            or f"Measures fraction of structures matching target value {str(target_value)}",
+            lower_is_better=lower_is_better,
+            n_jobs=n_jobs,
+        )
+        self.config = DiscreteTargetConfig(
+            name=self.config.name,
+            description=self.config.description,
+            lower_is_better=self.config.lower_is_better,
+            n_jobs=self.config.n_jobs,
+            target_value=target_value,
+        )
+
+    def _get_compute_attributes(self) -> dict[str, Any]:
+        """Get the attributes for the compute_structure method."""
+        return {
+            "cls": self.__class__,
+        }
+
+    @staticmethod
+    def value_extractor(structure: Structure) -> Any:
+        """Extract the relevant discrete value from a structure.
+
+        This method must be implemented by subclasses.
+        """
+        raise NotImplementedError("Subclasses must implement value_extractor")
+
+    @staticmethod  # TODO: Should this be a classmethod?
+    def compute_structure(structure: Structure, cls, **kwargs) -> float:
+        """Computes and returns the extracted value from the structure.
+        Returns None if value extraction fails.
+        """
+        value = cls.value_extractor(structure, **kwargs)
+        return float(value) if value is not None else np.nan
+
+    def aggregate_results(self, values: list[float | None]) -> Dict[str, Any]:
+        """Aggregate results into final metric values."""
+        valid_values = [v for v in values if v is not None]
```

Comment:
Although not critical, I would consider using arrays rather than lists in such operations.

```python
values = np.array(values, dtype=float)
valid_values = values[~np.isnan(values)]
...
success_values = (valid_values == self.config.target_value).sum()
success_rate = success_values.mean()
... # etc.
```